From: Eike Hein <hein@kde.org>
Date: Sun, 18 Dec 2016 08:30:46 +0900
Subject: Fix logic for arrow handling in RTL locales.

Summary: BUG:373749

Reviewers: #plasma, mart

Subscribers: plasma-devel

Tags: #plasma

Differential Revision: https://phabricator.kde.org/D3718
---
 .../plasmacomponents/qml/private/TabBarLayout.qml          | 14 ++++++--------
 1 file changed, 6 insertions(+), 8 deletions(-)

diff --git a/src/declarativeimports/plasmacomponents/qml/private/TabBarLayout.qml b/src/declarativeimports/plasmacomponents/qml/private/TabBarLayout.qml
index 49f08a7b2..0d75ccfe5 100644
--- a/src/declarativeimports/plasmacomponents/qml/private/TabBarLayout.qml
+++ b/src/declarativeimports/plasmacomponents/qml/private/TabBarLayout.qml
@@ -74,14 +74,12 @@ Item {
     onHeightChanged: layoutTimer.restart()
 
     Keys.onPressed: {
-        if (event.key == Qt.Key_Right || event.key == Qt.Key_Left) {
-            if (event.key == Qt.Key_Right || priv.mirrored) {
-                priv.goNextTab()
-                event.accepted = true
-            } else if (event.key == Qt.Key_Left || priv.mirrored) {
-                priv.goPreviousTab()
-                event.accepted = true
-            }
+        if (event.key == Qt.Key_Right) {
+            (priv.mirrored ? priv.goPreviousTab : priv.goNextTab)();
+            event.accepted = true
+        } else if (event.key == Qt.Key_Left || priv.mirrored) {
+            (priv.mirrored ? priv.goNextTab : priv.goPreviousTab)();
+            event.accepted = true
         }
     }
 
